lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -647,16 +701,18 @@ DiagnosticVerifier::verifyFile(unsigned BufferID, bool shouldAutoApplyFixes) {
   bool HadUnexpectedDiag = false;
   for (unsigned i = CapturedDiagnostics.size(); i != 0; ) {
```

Comment:
Oof, I know this isn't your bug, but we should use a modern iterator idiom here.